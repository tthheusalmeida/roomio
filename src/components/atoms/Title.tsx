import { mergeClassNames } from "@/utils/classNames";

interface TitleProps {
  children: React.ReactNode;
  className?: string;
}

export default function Title({ children, className }: TitleProps) {
  return (
    <>
      <h1 className={mergeClassNames("font-bold text-5xl my-8", className)}>
        {children}
      </h1>
    </>
  );
}
