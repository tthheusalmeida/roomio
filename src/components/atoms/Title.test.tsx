import { render, screen } from "@testing-library/react";
import Title from "./Title";

describe("Title component", () => {
  it("renders the children text correctly", () => {
    const { asFragment } = render(<Title>Main Title</Title>);

    const heading = screen.getByText("Main Title");
    expect(heading).toBeInTheDocument();
    expect(heading.tagName).toBe("H1");

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <Title className="extra-class">Main Title</Title>
    );

    const heading = screen.getByText("Main Title");
    expect(heading).toHaveClass("extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
