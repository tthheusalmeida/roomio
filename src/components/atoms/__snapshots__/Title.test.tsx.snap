// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Title component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <h1
    class="font-bold text-5xl my-8 extra-class"
  >
    Main Title
  </h1>
</DocumentFragment>
`;

exports[`Title component renders the children text correctly 1`] = `
<DocumentFragment>
  <h1
    class="font-bold text-5xl my-8"
  >
    Main Title
  </h1>
</DocumentFragment>
`;
